app/src/main/java/net/skyscanner/backpack/demo/compose/MapStory.kt

```diff
@@ -0,0 +1,161 @@
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.CameraPositionState
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.rememberMarkerState
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.map.BpkIconMapMarker
+import net.skyscanner.backpack.compose.map.BpkIconMarkerStatus
+import net.skyscanner.backpack.compose.map.BpkPointerMapMarker
+import net.skyscanner.backpack.compose.map.BpkPriceMapMarker
+import net.skyscanner.backpack.compose.map.BpkPriceMarkerStatus
+import net.skyscanner.backpack.compose.tokens.Cafe
+import net.skyscanner.backpack.compose.tokens.Landmark
+import net.skyscanner.backpack.demo.R
+import net.skyscanner.backpack.demo.components.MapMarkersComponent
+import net.skyscanner.backpack.demo.meta.ComposeStory
+import net.skyscanner.backpack.meta.StoryKind
+
+@Composable
+@MapMarkersComponent
+@ComposeStory(kind = StoryKind.DemoOnly, name = "Price")
+fun PriceMapMarkerStory(modifier: Modifier = Modifier) {
+    var focusedMarker by remember { mutableStateOf(0) }
+    var viewedMarkers by remember { mutableStateOf(setOf(1)) }
+
+    fun markerStatus(index: Int): BpkPriceMarkerStatus {
+        return when {
+            index == focusedMarker -> BpkPriceMarkerStatus.Focused
+            viewedMarkers.contains(index) -> BpkPriceMarkerStatus.Viewed
+            else -> BpkPriceMarkerStatus.Default
+        }
+    }
+
+    GoogleMap(cameraPositionState = CameraPositionState(position = MapPosition)) {
+        BpkPriceMapMarker(
+            title = stringResource(R.string.map_marker_price_1),
+            status = markerStatus(0),
+            state = rememberMarkerState(position = MarkerPosition1),
+            onClick = { focusedMarker = 0; viewedMarkers += 0 },
+        )
+
+        BpkPriceMapMarker(
+            title = stringResource(R.string.map_marker_price_2),
+            status = markerStatus(1),
+            state = rememberMarkerState(position = MarkerPosition2),
+            onClick = { focusedMarker = 1; viewedMarkers += 1 },
+        )
+
+        BpkPriceMapMarker(
+            title = stringResource(R.string.map_marker_price_3),
+            status = markerStatus(2),
+            state = rememberMarkerState(position = MarkerPosition3),
+            onClick = { focusedMarker = 2; viewedMarkers += 2 },
+        )
+
+        BpkPriceMapMarker(
+            title = stringResource(R.string.map_marker_price_4),
+            status = BpkPriceMarkerStatus.Disabled,
+            state = rememberMarkerState(position = MarkerPosition4),
+        )
+    }
+}
+
+@Composable
+@MapMarkersComponent
+@ComposeStory(kind = StoryKind.DemoOnly, name = "Icon")
+fun IconMapMarkerStory(modifier: Modifier = Modifier) {
+    var focusedMarker by remember { mutableStateOf(0) }
+    GoogleMap(cameraPositionState = CameraPositionState(position = MapPosition)) {
+        BpkIconMapMarker(
```

Comment:
I think this can be done as a for loop especially if you combine the MarkerPositions intro to a list. You might need to wrap the BpkIconMapBarker with `key(index)` though.

Reply:
it's not just the positions, though - the icon/state is also different